src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -0,0 +1,232 @@
+	}
+
+	isRuleKeySupported(): boolean {
+		// Javascript rules written for only eslint work for typescript as well,
+		// though they require the parser information that is specific to typescript.
+		return true;
+	}
+
+	getCatalogConfig(): Record<string, any> {
+		return ES_PLUS_TS_CONFIG;
+	}
+
+	getLanguage(): string[] {
+		return TypescriptEslintStrategy.LANGUAGE;
+	}
+
+	async getTargetPatterns(target?: string): Promise<string[]> {
+		const engineConfig = this.config.getEngineConfig(this.getName());
+
+		
+		// Find the typescript config file, if any
+		const tsconfigPath = await this.findTsconfig(target);
+
+		const targetPatterns: string[] = [];
+		if (tsconfigPath) {
+			const tsconfig: TSConfig = JSON.parse(await this.fileHandler.readFile(tsconfigPath));
+
+			// Found a tsconfig.  Load up its patterns.
+			if (tsconfig.include) {
+				targetPatterns.push(...tsconfig.include);
+			}
+
+			if (tsconfig.exclude) {
+				// Negate the exclude pattern (because that's how we like it)
+				targetPatterns.push(...tsconfig.exclude.map(e => "!" + e));
+			} else if (tsconfig.compilerOptions && tsconfig.compilerOptions.outDir) {
+				// TS likes to auto-exclude the outDir but only if exclude is not specified.
+				targetPatterns.push("!" + path.join(tsconfig.compilerOptions.outDir, "**"));
+			} else if (tsconfig.compilerOptions && tsconfig.compilerOptions.outFile) {
+				// Same reasoning as outDir
+				targetPatterns.push("!" + tsconfig.compilerOptions.outFile);
+			}
+
+			if (tsconfig.files) {
+				targetPatterns.push(...tsconfig.files);
+			}
+		}
+
+		// Join forces.  TODO or maybe not?  Should we just rely only on tsconfig if provided?
+		return engineConfig.targetPatterns.concat(targetPatterns);
+	}
+
+	filterUnsupportedPaths(paths: string[]): string[] {
+		const filteredPaths = paths.filter(p => p.endsWith(".ts"));
+		this.logger.trace(`Input paths: ${paths}, Filtered paths: ${filteredPaths}`);
+		return filteredPaths;
+	}
+
+	async getRunConfig(target?: string): Promise<Record<string, any>> {
+		const tsconfigPath = await this.findTsconfig(target);
+
+		if (!tsconfigPath) {
+			throw new Error(`Unable to find ${TS_CONFIG}. Please provide ${TS_CONFIG} in the target ${target} or the current working directory or in Config.`);
+		}
+
+		// Alert the user we found a config file, if --verbose
+		this.alertUser(tsconfigPath);
+
+		const config = {};
+
+		Object.assign(config, ES_PLUS_TS_CONFIG);
+
+		// Enable typescript by registering its project config file
+		config["parserOptions"].project = tsconfigPath;
+
+		this.logger.trace(`Using config for run: ${JSON.stringify(config)}`);
+
+		return config;
+	}
+
+	private alertUser(tsconfigPath: string): void {
+		const event = {
+			messageKey: 'info.usingEngineConfigFile', args: [tsconfigPath], type: 'INFO', handler: 'UX', verbose: true, time: Date.now()
+		};
+		this.outputProcessor.emitEvents([event]);
+	}
+
+	async findTsconfig(target: string): Promise<string> {
```

Comment:
This is the place where we control the priority of tsconfig.json file.
Config.json > target directory > current working directory